### PR TITLE
Adding {{t 'foo.bar'}} and {{t (l '{product} launched on {date, date, time})}}

### PR DESCRIPTION
**Pull request by jasonmit**
Todo:
* [x] Update Docs
* [x] Update tests
* [x] Add tests for `t` and `l`

```hbs
{{t 'product.info'
    product='Apple watch'
    price=200
    deadline=yesterday}}

{{t computedProperty
    product='Apple watch'
    price=200
    deadline=yesterday}}

Passing a string literal with the `l` helper

{{t (l '{name} took {numPhotos, plural,\n  =0 {no photos}\n  =1 {one photo}\n  other {# photos}\n} on {takenDate, date, long}')
    name='Jason'
    numPhotos=num
    takenDate=yesterday}}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -189,7 +189,7 @@ A new feature, only available when using >= Ember 1.13, is the ability to recomp
 [List of supported format date options](https://github.com/yahoo/ember-intl/wiki/Format-Relative-Options)
 
 ### Format Message
-Formats [ICU Message][ICU] strings with the given values supplied as the hash arguments.
+Formats [ICU Message][ICU] strings with the given values supplied as the hash arguments.  A short-hand form of the `{{format-message}}` is `{{t}}`.  They are aliases and behave identical.
```
Comment (elwayman02) [inline README.md]: identically

File: README.md [hunk 2]
```diff
@@ -199,12 +199,12 @@ You have {numPhotos, plural,
 ```
 
 ```hbs
-{{format-message (intl-get 'product.info')
+{{t 'product.info'
```
Comment (elwayman02) [inline README.md]: Are in intending to deprecate `format-message` and move to just `t` going forward? If not, I'd keep at least this example with the intl-get somewhere in the docs so people know how to use it should they choose.

Comment (elwayman02) [inline README.md]: Nvm, I see that intl-get is deprecated below

Comment (jasonmit) [inline README.md]: no reason to use intl-get, that was the point of this PR

**Pull request merged**